rust/datafusion/src/physical_plan/parquet.rs

```diff
@@ -209,6 +251,479 @@ impl ParquetExec {
+                    expr_builder.column_expr(),
+                    expr_builder.column_name(),
+                    max_column_name.as_str(),
+                )?))
+        }
+        Operator::Gt => {
+            let max_column_name = expr_builder.add_max_column();
+            // column > literal => (min, max) > literal => max > literal
+            rewrite_column_expr(
+                expr_builder.column_expr(),
+                expr_builder.column_name(),
+                max_column_name.as_str(),
+            )?
+            .gt(expr_builder.scalar_expr().clone())
+        }
+        Operator::GtEq => {
+            // column >= literal => (min, max) >= literal => max >= literal
+            let max_column_name = expr_builder.add_max_column();
+            rewrite_column_expr(
+                expr_builder.column_expr(),
+                expr_builder.column_name(),
+                max_column_name.as_str(),
+            )?
+            .gt_eq(expr_builder.scalar_expr().clone())
+        }
+        Operator::Lt => {
+            // (input < input) => (predicate)
+            // column < literal => (min, max) < literal => min < literal
+            let min_column_name = expr_builder.add_min_column();
+            rewrite_column_expr(
+                expr_builder.column_expr(),
+                expr_builder.column_name(),
+                min_column_name.as_str(),
+            )?
+            .lt(expr_builder.scalar_expr().clone())
+        }
+        Operator::LtEq => {
+            // (input <= input) => (predicate)
+            // column <= literal => (min, max) <= literal => min <= literal
+            let min_column_name = expr_builder.add_min_column();
+            rewrite_column_expr(
+                expr_builder.column_expr(),
+                expr_builder.column_name(),
+                min_column_name.as_str(),
+            )?
+            .lt_eq(expr_builder.scalar_expr().clone())
+        }
+        // other expressions are not supported
+        _ => {
+            return Ok(expressions::lit(ScalarValue::Boolean(Some(true))));
+        }
+    };
+    expr_builder.build(&statistics_expr)
+}
+
+/// replaces a column with an old name with a new name in an expression
+fn rewrite_column_expr(
+    expr: &Expr,
+    column_old_name: &str,
+    column_new_name: &str,
+) -> Result<Expr> {
+    let expressions = utils::expr_sub_expressions(&expr)?;
+    let expressions = expressions
+        .iter()
+        .map(|e| rewrite_column_expr(e, column_old_name, column_new_name))
+        .collect::<Result<Vec<_>>>()?;
+
+    if let Expr::Column(name) = expr {
+        if name == column_old_name {
+            return Ok(Expr::Column(column_new_name.to_string()));
+        }
+    }
+    utils::rewrite_expression(&expr, &expressions)
+}
+
+#[derive(Debug, Clone, PartialEq)]
+enum StatisticsType {
+    Min,
+    Max,
+}
+
+fn build_null_array(data_type: &DataType, length: usize) -> ArrayRef {
```

Comment:
I wonder if you could use `NullArray` here instead: https://github.com/apache/arrow/blob//rust/arrow/src/array/null.rs

Reply:
that's what I thought at first, and then realized that `NullArray` returns data type of `DataType::Null`, which doesn't work when the statistics record batch is created as it checks that types from the schema fields and from arrays are the same; that's why I wrote the `build_null_array` function

Reply:
makes sense

Reply:
@alamb  I have now changed the code to use `NullArray` but have had to add a new `new_with_type` constructor function (for the reason explained in my previous comment)